### ISSUE TITLE
prefixQueryOptions type value cannot be set

### ISSUE THREAD

**Issue by mledom** [searchkit/searchkit#445]
I'm new to running test for these packages so I provided the test case that shows the issue. I'm trying to set the type for the multiquery options to `cross_fields` instead of `phrase_prefix`. 
```
<SearchBox prefixQueryOptions={{type:'cross_fields'}}
  prefixQueryFields={['sup.firstName.prefix', 'sup.lastName.prefix']}
  queryBuilder={SimpleQueryString}
  .../>
```
Based on the current order of the params to assigns its not able to be set. Here is a testcase that I think should show the bug
```
it("prefixQueryFields with options change type", ()=> {
    this.accessor = new QueryAccessor("q", {
      prefixQueryFields:["title"],
      prefixQueryOptions: {
        type:"cross_fields"
      }
    })
    let query = new ImmutableQuery()
    this.accessor.state = new ValueState("some query")
    query = this.accessor.buildSharedQuery(query)
    expect(query.query.query).toEqual(BoolMust([
      BoolShould([
        SimpleQueryString("some query", {fields:["_all"]}),
        MultiMatchQuery("some query", {
          type:"cross_fields",
          fields:["title"]
        })
      ])
    ]))
  })
```

**Comment by TheDolo** [comment]
I could be wrong, but SimpleQueryString does not offer cross_fields or support prefixQueryOptions?

Per the authors docs:
prefixQueryOptions (Object): Optional. An object of options for MultiMatchQuery .

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html

**Comment by mledom** [comment]
I did end up doing that, that's how I knew that I wanted to use the cross_fields type. When I attempted to use that option with prefixQueryOptions it would not get set. I looked into the code and found on this line: [error line](https://github.com/searchkit/searchkit/blob/f59f28398928c47ef0ec3b25706c224b008766b5/src/core/accessors/QueryAccessor.ts#L50) the parameter for assigns passes the default option as the second option vs. this line: [working line](https://github.com/searchkit/searchkit/blob/f59f28398928c47ef0ec3b25706c224b008766b5/src/core/accessors/QueryAccessor.ts#L41) where the default is the first parameter. According the the docs (below) when passing the defaults as the second option they will overwrite the previous value in the options hash set by the user:

>The _.assign function (also called _.extend) accepts as its first parameter a target object to which all properties of the following parameters will be assigned. Values defined by later parameters will overwrite sooner values.
>

I was able to update my version of the code to see if working but was unsure how to run the test suite for searchkit.
```        
queries.push(MultiMatchQuery(queryStr, assign(
  {
    type:"phrase_prefix",
    fields:this.options.prefixQueryFields,
  }, this.options.prefixQueryOptions))
)
```
